src/app/src/main/kotlin/org/vaccineimpact/api/app/repositories/jooq/JooqTouchstoneRepository.kt

```diff
@@ -191,14 +191,80 @@ class JooqTouchstoneRepository(
         return records.map { mapCoverageSet(it) }
     }
 
+    private fun coverageDimensions() : Array<Field<*>>
+    {
+        //The columns in the Coverage table which the target and coverage values are grouped by
+        return arrayOf(COVERAGE.COVERAGE_SET,
+                COVERAGE.YEAR,
+                COVERAGE.COUNTRY,
+                COVERAGE.AGE_FROM,
+                COVERAGE.AGE_TO,
+                COVERAGE.AGE_RANGE_VERBATIM,
+                COVERAGE.GAVI_SUPPORT,
+                COVERAGE.GENDER)
+    }
+
+    private fun aggregatedValues() : List<Field<*>>
+    {
+        return arrayOf(
+                roundValueIfRequired(::aggregatedCoverage).`as`("coverage"),
+                roundValueIfRequired(::aggregatedTarget).`as`("target")
+        ).toList()
+    }
+
+    private fun roundValueIfRequired(valueFun : () -> Field<BigDecimal?>) : Field<BigDecimal?>
```

Comment:
It occurred to me that we could maybe use a user-defined function to calculate these rounded values - in a function you could store the aggregated value result in a declared variable so you wouldn't need to keep recalculating it in a single SQL expression. Not sure if we're already using UDFs elsewhere? Of course they have a maintenance overhead, and potentially a performance overhead of their own. I assume you can use a function on all the rows of a result set? You could in SQL Server. Maybe one to try if performance of this solution is poor...